tsinfer/formats.py

```diff
@@ -766,6 +781,10 @@ class SampleData(DataContainer):
     ADDING_SAMPLES = 1
     ADDING_SITES = 2
 
+    # Marker constants for time value
+    TIME_UNSPECIFIED = -np.inf  # Placeholder to mark freq of derived alleles as time
+    TIME_MEANINGLESS = np.inf  # Placeholder for e.g. freq value for a nonvariable site
```

Comment:
We're still making judgements here about what the time **means**. I think we should just have TIME_UNSPECIFIED and nothing else. Leave the semantics to the consumer.

Reply:
I see "unspecified by user" as quite different to "algorithmically meaningless". The first case is used to store the user preference. The second is to flag up times when we wrongly try to use a meaningless time. It's demonstrated by the fact that by adding `TIME_MEANINGLESS` in addition to `TIME_UNSPECIFIED`, I managed to pick up a long standing bug in the test suite.

Reply:
Also, if we programatically set `TIME_UNSPECIFIED` when we really mean `TIME_MEANINGLESS`, we override any information about what the user originally fed in to the algorithm. I'm prepared to say that we should kill off `TIME_MEANINGLESS` and simply put `np.nan` in there though, which is what we would get anyway when all samples at a site are marked as missing data, and the freq is 0/0.